plugins/cilium-cni/cmd/cmd.go

```diff
@@ -665,6 +692,13 @@ func (cmd *Cmd) Add(args *skel.CmdArgs) (err error) {
 			if err != nil {
 				return fmt.Errorf("unable to setup interface datapath: %w", err)
 			}
+		case ipamOption.IPAMDelegatedPlugin:
+			if conf.InstallUplinkRoutesForDelegatedIPAM {
```

Comment:
nit: Could we extract a helper function and replace the whole switch statement with an `if needsEndpointRoutingRule(conf) { interfaceAdd(...) }`?